storage/innobase/row/row0merge.cc

```diff
@@ -1060,83 +1061,75 @@ static dberr_t row_merge_write_blob_to_tmp_file(
       return DB_OUT_OF_MEMORY;
   }
   uint64_t val= blob_file->offset;
-  uint32_t len= field->len;
   dberr_t err= os_file_write(
     IORequestWrite, "(bulk insert)", blob_file->fd,
-    field->data, blob_file->offset, len);
+    (byte*)field->data + local_len, blob_file->offset,
+    field->len - local_len);
 
   if (err != DB_SUCCESS)
     return err;
 
-  byte *data= static_cast<byte*>
-    (mem_heap_alloc(*heap, BTR_EXTERN_FIELD_REF_SIZE));
+  byte *data=
+    static_cast<byte*>(mem_heap_alloc(*heap,
+                                      local_len + BTR_EXTERN_FIELD_REF_SIZE));
 
+  memcpy(data, field->data, local_len);
   /* Write zeroes for first 8 bytes */
-  memset(data, 0, 8);
+  memset(data + local_len, 0, 8);
   /* Write offset for next 8 bytes */
-  mach_write_to_8(data + 8, val);
+  mach_write_to_8(data + local_len + 8, val);
   /* Write length of the blob in 4 bytes */
-  mach_write_to_4(data + 16, len);
-  blob_file->offset+= field->len;
+  mach_write_to_4(data + local_len + 16, field->len - local_len);
+  blob_file->offset+= (field->len - local_len);
   blob_file->n_rec++;
-  dfield_set_data(field, data, BTR_EXTERN_FIELD_REF_SIZE);
+  dfield_set_data(field, data,
+                  local_len + BTR_EXTERN_FIELD_REF_SIZE);
   dfield_set_ext(field);
```

Comment:
We could invoke `dfield_set_data()` and `dfield_set_ext()` at the start and then do `data += local_len` to reduce the diff and to possibly simplify the code, in case the common subexpression elimination (CSE) phase of the compiler might miss the repeated `data + local_len`.